### ISSUE TITLE
|| iterator generates invalid code when compiling with --debugger:native

### ISSUE THREAD

**Issue by krux02** [nim-lang/Nim#9710]
``nim c -r --debugger:native scratch.nim``

```nim
for i in 1 || 200:
  echo i
```

### Current Output

```
CC: scratch
CC: stdlib_system
Error: execution of an external compiler program 'clang -c -w -g  -I/home/arne/proj/nim/Nim/lib -o /home/arne/.cache/nim/scratch_d/scratch.c.o /home/arne/.cache/nim/scratch_d/scratch.c' failed with exit code: 1

/tmp/scratch.nim:1:28: error: expected expression
        nimln_(1, "scratch.nim");       #pragma omp parallel for
                                        ^
1 error generated.

Compilation exited abnormally with code 1 at Wed Nov 14 15:26:29
```

**Comment by ringabout** [comment]
It has been fixed since Nim 1.0

```nim
for i in 1 || 200:
  echo i
```

Output:
1 ...... 200

**Comment by Clyybber** [comment]
Closed with #15365
